src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/evosuite/Runner.kt

```diff
@@ -79,32 +91,74 @@ class Runner(private val projectPath: String, private val projectClassPath: Stri
         log.info("Starting EvoSuite with arguments: $cmdString")
         log.info("Results will be saved to $serializeResultPath")
 
-        Thread {
-            val evoSuiteProcess = GeneralCommandLine(cmd)
-            evoSuiteProcess.charset = Charset.forName("UTF-8")
-            evoSuiteProcess.setWorkDirectory(projectPath)
 
-            val handler: ProcessHandler = OSProcessHandler(evoSuiteProcess)
-
-            val outputBuilder = StringBuilder()
-
-            // attach process listener for output
-            handler.addProcessListener(object : ProcessAdapter() {
-                override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
-                    val text = event.text
-                    outputBuilder.append(text)
-                    log.info(text)
+        ProgressManager.getInstance()
+            .run(object : Task.Backgroundable(project, TestGenieBundle.message("evosuiteTestGenerationMessage")) {
+                override fun run(indicator: ProgressIndicator) {
+                    indicator.isIndeterminate = false
+                    indicator.text = TestGenieBundle.message("evosuiteSearchMessage")
```

Comment:
this `indicator`, I see you append a lot of info to it. Does it print/log this somewhere? Or is this more of an "internal metric" you use in the code?

Reply:
`ProgressIndicator` represents the state of the bar you see when you open a background task in intellij.